benchmarks/map-reduce/knative_yamls/s3/service-driver.yaml

```diff
@@ -42,9 +42,15 @@ spec:
             - name: TRANSFER_TYPE
               value: "S3"
             - name: AWS_ACCESS_KEY
-              value: ${AWS_ACCESS_KEY}
+              valueFrom:
+                secretKeyRef:
+                  name: awscreds
+                  key: awsaccess
             - name: AWS_SECRET_KEY
-              value: ${AWS_SECRET_KEY}
+              valueFrom:
+                secretKeyRef:
+                  name: awscreds
+                  key: awssecret
```

Comment:
Currently in the _main_ branch, `AWS_ACCESS_KEY` and `AWS_SECRET_KEY` are passed as environment variables like this:
```
- name: AWS_ACCESS_KEY
- value: ${AWS_ACCESS_KEY}
```
This fails to pass the environment variable to the knative service. If you try printing `os.environ.get('AWS_ACCESS_KEY')` then you will find that it is equal to the string "${AWS_ACCESS_KEY}" and not the actual value of that variable.
I think the right way would be to just specify the environment variable as _$AWS_ACCESS_KEY_ without the curly braces as shown [here](https://stackoverflow.com/a/56009991).

Kubernetes provides a way to create and pass secrets to services. You first define the secrets using `kubectl create secret` (see [here](https://github.com/vhive-serverless/vSwarm/blob/aws_lambda_support/.github/workflows/e2e-map-reduce.yml#L179)). Then you pass them to the service in the way you see.

I adopted this way because this is the recommended and secure method of passing sensitive data to knative services.